### PR TITLE
ROC-4139 Add case metadata endpoints to search by claimant and defendant emails

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-4139

### Change description ###
New endpoints added to the case metadata controller to support search by email.
Both claimant and defendant email searches are available.
POST verb is used to keep user emails out of logs.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/cmc/claimstore/controllers/CaseMetadataControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    public void shouldThrowNotFoundExceptionWhenNoClaimWithReference() {
+        // given
+        when(claimService.getClaimByReferenceAnonymous("reference"))
+            .thenReturn(Optional.empty());
+
+        // when
+        CaseMetadata output = controller.getByClaimReference("reference");
+
+        // then exception should have been thrown
+        fail("Expected exception was not thrown; returned instead: " + output);
```
Comment (timja) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/controllers/CaseMetadataControllerTest.java]: is this needed?

Comment (andrewwa-kainos) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/controllers/CaseMetadataControllerTest.java]: Nope

File: src/test/java/uk/gov/hmcts/cmc/claimstore/controllers/CaseMetadataControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CaseMetadataControllerTest {
```
Comment (timja) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/controllers/CaseMetadataControllerTest.java]: I would recommend converting this to an api or functional test, it doesn't serve much value imo

Comment (andrewwa-kainos) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/controllers/CaseMetadataControllerTest.java]: Added api tests - want me to remove this as a target of unnecessary maintenance or just keep it for the coverage stats?

Comment (timja) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/controllers/CaseMetadataControllerTest.java]: api tests should already be included in coverage

**Comment by andrewwa-kainos** [comment]
For testing:
curl command is
`curl --data "email=andrewwa@kainos.com" http://localhost:4400/claims/filters/claimants/email | jq`
or with httpie
`http -f :4400/claims/filters/claimants/email email=andrewwa@kainos.com`
(Will add to confluence when merged)

**Pull request merged**